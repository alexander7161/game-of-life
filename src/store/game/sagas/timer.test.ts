import { advanceGame, pauseGame, startGame } from "..";
import timerSaga from "./timer";
import SagaTester from "redux-saga-tester";
import GameReducer from "..";
describe("timer", () => {
  it("Should start timer on StartGame", async () => {
    const sagaTester = new SagaTester({
      reducers: { game: GameReducer },
    });
    sagaTester.start(timerSaga);

    sagaTester.dispatch(startGame());
    await sagaTester.waitFor(advanceGame.toString(), true);
    await sagaTester.waitFor(advanceGame.toString(), true);
    await sagaTester.waitFor(advanceGame.toString(), true);
    sagaTester.dispatch(pauseGame());
    expect(sagaTester.getCalledActions()).toMatchSnapshot();
  });

  it("Should stop timer on pauseGame", async () => {
    const sagaTester = new SagaTester({ reducers: { game: GameReducer } });
    sagaTester.start(timerSaga);

    sagaTester.dispatch(startGame());
    await sagaTester.waitFor(advanceGame.toString());
    sagaTester.dispatch(pauseGame());
    expect(sagaTester.numCalled(advanceGame.toString())).toBe(1);
    expect(sagaTester.getCalledActions()).toMatchSnapshot();
  });
});
