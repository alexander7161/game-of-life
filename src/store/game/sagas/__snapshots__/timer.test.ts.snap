// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`timer Should start timer on StartGame 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "game/startGame",
  },
  Object {
    "payload": undefined,
    "type": "game/advanceGame",
  },
  Object {
    "payload": undefined,
    "type": "game/advanceGame",
  },
  Object {
    "payload": undefined,
    "type": "game/advanceGame",
  },
  Object {
    "payload": undefined,
    "type": "game/pauseGame",
  },
]
`;

exports[`timer Should stop timer on pauseGame 1`] = `
Array [
  Object {
    "payload": undefined,
    "type": "game/startGame",
  },
  Object {
    "payload": undefined,
    "type": "game/advanceGame",
  },
  Object {
    "payload": undefined,
    "type": "game/pauseGame",
  },
]
`;
